### PR TITLE
Added parameter to hide option for calendar subscription via link

### PR DESCRIPTION
**Pull request by pboguslawski**
Added calendar application config parameter `allow_subscribe_link` to hide option for
calendar subscription via link  (i.e. for internal Nextcloud setups). Use

```
occ config:app:set calendar allow_subscribe_link --value 'yes'
```

to show (default if not set) and

```
occ config:app:set calendar allow_subscribe_link --value 'no'
```

to hide calendar subscription via link option. Should be applied together with
https://github.com/nextcloud/server/pull/34387

Related: https://github.com/nextcloud/server/pull/34387
Author-Change-Id: IB#1126265
Signed-off-by: Pawel Boguslawski <pawel.boguslawski@ib.pl>

### CODE REVIEW

**Comment by miaulalala** [comment]
Hi, thanks for your PR! Could you please fix the tests that are failing, then I'll review.

**Comment by pboguslawski** [comment]
> Hi, thanks for your PR! Could you please fix the tests that are failing, then I'll review.

Done in 0baaebc69aafe8ab34025f66b02f77fa9e89759b.

File: lib/Controller/PublicViewController.php [hunk 1]
```diff
@@ -128,6 +129,7 @@ private function publicIndex(string $token,
 		$defaultSlotDuration = $this->config->getAppValue($this->appName, 'slotDuration', '00:30:00');
 		$defaultDefaultReminder = $this->config->getAppValue($this->appName, 'defaultReminder', 'none');
 		$defaultShowTasks = $this->config->getAppValue($this->appName, 'showTasks', 'yes');
+		$defaultCanSubscribeLink = $this->config->getAppValue('dav', 'allow_calendar_link_subscriptions', 'yes');
```
Comment (tcitworld) [inline lib/Controller/PublicViewController.php:132]:
```suggestion
		$defaultCanSubscribeLink = $this->config->getAppValue('dav', 'allow_calendar_link_subscriptions', 'yes');
```

**Review by tcitworld — APPROVED** [review]
We could have done this by checking the available server features through an OPTIONS request, but this is simpler.

**Pull request merged**